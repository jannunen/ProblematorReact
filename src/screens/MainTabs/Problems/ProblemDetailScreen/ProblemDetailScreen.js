import React, { Component } from 'react'
import {
    View,
    Image,
    Text,
    Button,
    StyleSheet,
    TouchableOpacity,
    Platform,
    Dimensions
  } from "react-native";
import { connect } from "react-redux";
import Icon from "react-native-vector-icons/Ionicons";


export default class ProblemDetailScreen extends Component {

    constructor(props) {
        super(props);
        //Dimensions.addEventListener("change", this.updateStyles);
      }

    render() {
        return (
            <View>
            {this.props.problem.problemid}
            </View>
        );
    }
}