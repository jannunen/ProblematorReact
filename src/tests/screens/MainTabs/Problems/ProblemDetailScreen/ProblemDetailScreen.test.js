import 'react-native';
import React from 'react';
import sinon from 'sinon'
import configureMockStore from 'redux-mock-store'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Navigation} from 'react-native-navigation';
import thunk from 'redux-thunk'
import problem from '../../../../fixtures/problem';
import ProblemDetailScreen from '../../../../../screens/MainTabs/Problems/ProblemDetailScreen/ProblemDetailScreen';

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)


let  history, wrapper;

describe('ProblemDetailScreen tests',() => {

it('should correctly render ProblemDetailScreen', () => {
  const store = mockStore({ loading: false, error :false })
  const tree = shallow(<ProblemDetailScreen problem={problem} />);
  expect(tree).toMatchSnapshot();
});




})