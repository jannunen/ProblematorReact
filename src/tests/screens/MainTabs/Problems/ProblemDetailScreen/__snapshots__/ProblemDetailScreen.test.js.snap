// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProblemDetailScreen tests should correctly render ProblemDetailScreen 1`] = `
<ProblemDetails
  problem={
    Object {
      "added": "2018-12-27 14:03:16",
      "addedformatted": "27.12.2018",
      "addedrelative": "2 weeks ago",
      "addt": "",
      "attr_bodytension": "0",
      "attr_complex": "1",
      "attr_compression": "0",
      "attr_coordination": "0",
      "attr_crimpers": "0",
      "attr_dyno": "0",
      "attr_endurance": "0",
      "attr_fingery": "0",
      "attr_hooks": "0",
      "attr_jugs": "0",
      "attr_mantle": "1",
      "attr_pinches": "0",
      "attr_pockets": "0",
      "attr_powerful": "1",
      "attr_sensitive": "0",
      "attr_slopers": "0",
      "attr_technical": "0",
      "attr_volumes": "0",
      "australian": "24",
      "author": "Tuukka",
      "c_dangerous": "0",
      "c_dirty": "0",
      "c_dislike": "0",
      "c_like": "0",
      "c_love": "0",
      "chartcolor": "#B370CF",
      "code": "#000",
      "color133": "black",
      "colour": "Black",
      "colourcode": "#000",
      "colourid": "10",
      "crag_route_id": null,
      "enddefinition": "Last hold",
      "ext_id": null,
      "font": "7a",
      "gradeid": "11",
      "gradename": "7A",
      "gymid": "132",
      "hidedate": null,
      "htmlcode": "#000",
      "id": "10",
      "lastWall": "Släbi",
      "name": "Black",
      "problemid": "47428",
      "routetype": "boulder",
      "score": "700",
      "slider_complexity": "0",
      "slider_intensity": "0",
      "slider_risk": "0",
      "soontoberemoved": "0",
      "sort": null,
      "south_africa": null,
      "stars": null,
      "startdefinition": "One hold",
      "tag": "BKE2018A034",
      "tagshort": "A034",
      "tapecolour": "#FF0000",
      "textcolor": "#fff",
      "tick": null,
      "uiaa": "VIII",
      "visible": "1",
      "vscale": "V6",
      "wallchar": "A",
      "walldesc": "Släbi",
      "wallid": "770",
      "yds": "5.11d",
    }
  }
/>
`;
